From 2591c280c375c547022eb619895c1aebbbc38219 Mon Sep 17 00:00:00 2001
From: Joe Perches <joe@perches.com>
Date: Sun, 4 Sep 2016 14:24:03 -0700
Subject: [PATCH 20/20] qed: Remove OOM messages

These messages are unnecessary as OOM allocation failures already do
a dump_stack() giving more or less the same information.

$ size drivers/net/ethernet/qlogic/qed/built-in.o* (defconfig x86-64)
   text	   data	    bss	    dec	    hex	filename
 127817	  27969	  32800	 188586	  2e0aa	drivers/net/ethernet/qlogic/qed/built-in.o.new
 132474	  27969	  32800	 193243	  2f2db	drivers/net/ethernet/qlogic/qed/built-in.o.old

Miscellanea:

o Change allocs to the generally preferred forms where possible.

Signed-off-by: Joe Perches <joe@perches.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/ethernet/qlogic/qed/qed_cxt.c      | 20 +++------
 drivers/net/ethernet/qlogic/qed/qed_dcbx.c     | 13 ++----
 drivers/net/ethernet/qlogic/qed/qed_dev.c      | 60 +++++++-------------------
 drivers/net/ethernet/qlogic/qed/qed_hw.c       | 12 ++----
 drivers/net/ethernet/qlogic/qed/qed_init_ops.c |  4 +-
 drivers/net/ethernet/qlogic/qed/qed_int.c      | 23 +++-------
 drivers/net/ethernet/qlogic/qed/qed_main.c     |  4 +-
 drivers/net/ethernet/qlogic/qed/qed_mcp.c      |  1 -
 drivers/net/ethernet/qlogic/qed/qed_spq.c      | 28 +++---------
 drivers/net/ethernet/qlogic/qed/qed_sriov.c    |  9 ++--
 drivers/net/ethernet/qlogic/qed/qed_vf.c       | 14 ++----
 11 files changed, 47 insertions(+), 141 deletions(-)

Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_cxt.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_cxt.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_cxt.c
@@ -792,10 +792,9 @@ static int qed_cxt_src_t2_alloc(struct q
	p_mngr->t2_num_pages = DIV_ROUND_UP(total_size, psz);

	/* allocate t2 */
-	p_mngr->t2 = kzalloc(p_mngr->t2_num_pages * sizeof(struct qed_dma_mem),
+	p_mngr->t2 = kcalloc(p_mngr->t2_num_pages, sizeof(struct qed_dma_mem),
			     GFP_KERNEL);
	if (!p_mngr->t2) {
-		DP_NOTICE(p_hwfn, "Failed to allocate t2 table\n");
		rc = -ENOMEM;
		goto t2_fail;
	}
@@ -957,7 +956,6 @@ static int qed_ilt_shadow_alloc(struct q
	p_mngr->ilt_shadow = kcalloc(size, sizeof(struct qed_dma_mem),
				     GFP_KERNEL);
	if (!p_mngr->ilt_shadow) {
-		DP_NOTICE(p_hwfn, "Failed to allocate ilt shadow table\n");
		rc = -ENOMEM;
		goto ilt_shadow_fail;
	}
@@ -1050,10 +1048,8 @@ int qed_cxt_mngr_alloc(struct qed_hwfn *
	u32 i;

	p_mngr = kzalloc(sizeof(*p_mngr), GFP_KERNEL);
-	if (!p_mngr) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `struct qed_cxt_mngr'\n");
+	if (!p_mngr)
		return -ENOMEM;
-	}

	/* Initialize ILT client registers */
	clients = p_mngr->clients;
@@ -1105,24 +1101,18 @@ int qed_cxt_tables_alloc(struct qed_hwfn

	/* Allocate the ILT shadow table */
	rc = qed_ilt_shadow_alloc(p_hwfn);
-	if (rc) {
-		DP_NOTICE(p_hwfn, "Failed to allocate ilt memory\n");
+	if (rc)
		goto tables_alloc_fail;
-	}

	/* Allocate the T2  table */
	rc = qed_cxt_src_t2_alloc(p_hwfn);
-	if (rc) {
-		DP_NOTICE(p_hwfn, "Failed to allocate T2 memory\n");
+	if (rc)
		goto tables_alloc_fail;
-	}

	/* Allocate and initialize the acquired cids bitmaps */
	rc = qed_cid_map_alloc(p_hwfn);
-	if (rc) {
-		DP_NOTICE(p_hwfn, "Failed to allocate cid maps\n");
+	if (rc)
		goto tables_alloc_fail;
-	}

	return 0;

Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_dcbx.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_dcbx.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_dcbx.c
@@ -875,11 +875,8 @@ int qed_dcbx_info_alloc(struct qed_hwfn
	int rc = 0;

	p_hwfn->p_dcbx_info = kzalloc(sizeof(*p_hwfn->p_dcbx_info), GFP_KERNEL);
-	if (!p_hwfn->p_dcbx_info) {
-		DP_NOTICE(p_hwfn,
-			  "Failed to allocate 'struct qed_dcbx_info'\n");
+	if (!p_hwfn->p_dcbx_info)
		rc = -ENOMEM;
-	}

	return rc;
 }
@@ -1190,10 +1187,8 @@ int qed_dcbx_get_config_params(struct qe
	}

	dcbx_info = kzalloc(sizeof(*dcbx_info), GFP_KERNEL);
-	if (!dcbx_info) {
-		DP_ERR(p_hwfn, "Failed to allocate struct qed_dcbx_info\n");
+	if (!dcbx_info)
		return -ENOMEM;
-	}

	rc = qed_dcbx_query_params(p_hwfn, dcbx_info, QED_DCBX_OPERATIONAL_MIB);
	if (rc) {
@@ -1227,10 +1222,8 @@ static struct qed_dcbx_get *qed_dcbnl_ge
	struct qed_dcbx_get *dcbx_info;

	dcbx_info = kzalloc(sizeof(*dcbx_info), GFP_KERNEL);
-	if (!dcbx_info) {
-		DP_ERR(hwfn->cdev, "Failed to allocate memory for dcbx_info\n");
+	if (!dcbx_info)
		return NULL;
-	}

	if (qed_dcbx_query_params(hwfn, dcbx_info, type)) {
		kfree(dcbx_info);
Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_dev.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_dev.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_dev.c
@@ -340,7 +340,6 @@ static int qed_init_qm_info(struct qed_h
	return 0;

 alloc_err:
-	DP_NOTICE(p_hwfn, "Failed to allocate memory for QM params\n");
	qed_qm_info_free(p_hwfn);
	return -ENOMEM;
 }
@@ -424,18 +423,12 @@ int qed_resc_alloc(struct qed_dev *cdev)
				     RESC_NUM(p_hwfn, QED_L2_QUEUE);

		p_hwfn->p_tx_cids = kzalloc(tx_size, GFP_KERNEL);
-		if (!p_hwfn->p_tx_cids) {
-			DP_NOTICE(p_hwfn,
-				  "Failed to allocate memory for Tx Cids\n");
+		if (!p_hwfn->p_tx_cids)
			goto alloc_no_mem;
-		}

		p_hwfn->p_rx_cids = kzalloc(rx_size, GFP_KERNEL);
-		if (!p_hwfn->p_rx_cids) {
-			DP_NOTICE(p_hwfn,
-				  "Failed to allocate memory for Rx Cids\n");
+		if (!p_hwfn->p_rx_cids)
			goto alloc_no_mem;
-		}
	}

	for_each_hwfn(cdev, i) {
@@ -522,26 +515,18 @@ int qed_resc_alloc(struct qed_dev *cdev)

		/* DMA info initialization */
		rc = qed_dmae_info_alloc(p_hwfn);
-		if (rc) {
-			DP_NOTICE(p_hwfn,
-				  "Failed to allocate memory for dmae_info structure\n");
+		if (rc)
			goto alloc_err;
-		}

		/* DCBX initialization */
		rc = qed_dcbx_info_alloc(p_hwfn);
-		if (rc) {
-			DP_NOTICE(p_hwfn,
-				  "Failed to allocate memory for dcbx structure\n");
+		if (rc)
			goto alloc_err;
-		}
	}

	cdev->reset_stats = kzalloc(sizeof(*cdev->reset_stats), GFP_KERNEL);
-	if (!cdev->reset_stats) {
-		DP_NOTICE(cdev, "Failed to allocate reset statistics\n");
+	if (!cdev->reset_stats)
		goto alloc_no_mem;
-	}

	return 0;

@@ -1713,10 +1698,8 @@ static int qed_hw_prepare_single(struct

	/* Allocate PTT pool */
	rc = qed_ptt_pool_alloc(p_hwfn);
-	if (rc) {
-		DP_NOTICE(p_hwfn, "Failed to prepare hwfn's hw\n");
+	if (rc)
		goto err0;
-	}

	/* Allocate the main PTT */
	p_hwfn->p_main_ptt = qed_get_reserved_ptt(p_hwfn, RESERVED_PTT_MAIN);
@@ -1746,10 +1729,8 @@ static int qed_hw_prepare_single(struct

	/* Allocate the init RT array and initialize the init-ops engine */
	rc = qed_init_alloc(p_hwfn);
-	if (rc) {
-		DP_NOTICE(p_hwfn, "Failed to allocate the init array\n");
+	if (rc)
		goto err2;
-	}

	return rc;
 err2:
@@ -1957,10 +1938,8 @@ qed_chain_alloc_next_ptr(struct qed_dev
		p_virt = dma_alloc_coherent(&cdev->pdev->dev,
					    QED_CHAIN_PAGE_SIZE,
					    &p_phys, GFP_KERNEL);
-		if (!p_virt) {
-			DP_NOTICE(cdev, "Failed to allocate chain memory\n");
+		if (!p_virt)
			return -ENOMEM;
-		}

		if (i == 0) {
			qed_chain_init_mem(p_chain, p_virt, p_phys);
@@ -1990,10 +1969,8 @@ qed_chain_alloc_single(struct qed_dev *c

	p_virt = dma_alloc_coherent(&cdev->pdev->dev,
				    QED_CHAIN_PAGE_SIZE, &p_phys, GFP_KERNEL);
-	if (!p_virt) {
-		DP_NOTICE(cdev, "Failed to allocate chain memory\n");
+	if (!p_virt)
		return -ENOMEM;
-	}

	qed_chain_init_mem(p_chain, p_virt, p_phys);
	qed_chain_reset(p_chain);
@@ -2010,13 +1987,9 @@ static int qed_chain_alloc_pbl(struct qe
	void *p_virt = NULL;

	size = page_cnt * sizeof(*pp_virt_addr_tbl);
-	pp_virt_addr_tbl = vmalloc(size);
-	if (!pp_virt_addr_tbl) {
-		DP_NOTICE(cdev,
-			  "Failed to allocate memory for the chain virtual addresses table\n");
+	pp_virt_addr_tbl = vzalloc(size);
+	if (!pp_virt_addr_tbl)
		return -ENOMEM;
-	}
-	memset(pp_virt_addr_tbl, 0, size);

	/* The allocation of the PBL table is done with its full size, since it
	 * is expected to be successive.
@@ -2029,19 +2002,15 @@ static int qed_chain_alloc_pbl(struct qe
					size, &p_pbl_phys, GFP_KERNEL);
	qed_chain_init_pbl_mem(p_chain, p_pbl_virt, p_pbl_phys,
			       pp_virt_addr_tbl);
-	if (!p_pbl_virt) {
-		DP_NOTICE(cdev, "Failed to allocate chain pbl memory\n");
+	if (!p_pbl_virt)
		return -ENOMEM;
-	}

	for (i = 0; i < page_cnt; i++) {
		p_virt = dma_alloc_coherent(&cdev->pdev->dev,
					    QED_CHAIN_PAGE_SIZE,
					    &p_phys, GFP_KERNEL);
-		if (!p_virt) {
-			DP_NOTICE(cdev, "Failed to allocate chain memory\n");
+		if (!p_virt)
			return -ENOMEM;
-		}

		if (i == 0) {
			qed_chain_init_mem(p_chain, p_virt, p_phys);
@@ -2076,7 +2045,8 @@ int qed_chain_alloc(struct qed_dev *cdev
	rc = qed_chain_alloc_sanity_check(cdev, cnt_type, elem_size, page_cnt);
	if (rc) {
		DP_NOTICE(cdev,
-			  "Cannot allocate a chain with the given arguments:\n"
+			  "Cannot allocate a chain with the given arguments:\n");
+		DP_NOTICE(cdev,
			  "[use_mode %d, mode %d, cnt_type %d, num_elems %d, elem_size %zu]\n",
			  intended_use, mode, cnt_type, num_elems, elem_size);
		return rc;
Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_hw.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_hw.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_hw.c
@@ -482,28 +482,22 @@ int qed_dmae_info_alloc(struct qed_hwfn

	*p_comp = dma_alloc_coherent(&p_hwfn->cdev->pdev->dev,
				     sizeof(u32), p_addr, GFP_KERNEL);
-	if (!*p_comp) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `p_completion_word'\n");
+	if (!*p_comp)
		goto err;
-	}

	p_addr = &p_hwfn->dmae_info.dmae_cmd_phys_addr;
	*p_cmd = dma_alloc_coherent(&p_hwfn->cdev->pdev->dev,
				    sizeof(struct dmae_cmd),
				    p_addr, GFP_KERNEL);
-	if (!*p_cmd) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `struct dmae_cmd'\n");
+	if (!*p_cmd)
		goto err;
-	}

	p_addr = &p_hwfn->dmae_info.intermediate_buffer_phys_addr;
	*p_buff = dma_alloc_coherent(&p_hwfn->cdev->pdev->dev,
				     sizeof(u32) * DMAE_MAX_RW_SIZE,
				     p_addr, GFP_KERNEL);
-	if (!*p_buff) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `intermediate_buffer'\n");
+	if (!*p_buff)
		goto err;
-	}

	p_hwfn->dmae_info.channel = p_hwfn->rel_pf_id;

Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_init_ops.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_init_ops.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_init_ops.c
@@ -460,10 +460,8 @@ int qed_init_run(struct qed_hwfn *p_hwfn
	init_ops = cdev->fw_data->init_ops;

	p_hwfn->unzip_buf = kzalloc(MAX_ZIPPED_SIZE * 4, GFP_ATOMIC);
-	if (!p_hwfn->unzip_buf) {
-		DP_NOTICE(p_hwfn, "Failed to allocate unzip buffer\n");
+	if (!p_hwfn->unzip_buf)
		return -ENOMEM;
-	}

	for (cmd_num = 0; cmd_num < num_init_ops; cmd_num++) {
		union init_op *cmd = &init_ops[cmd_num];
Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_int.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_int.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_int.c
@@ -2370,10 +2370,8 @@ static int qed_int_sb_attn_alloc(struct

	/* SB struct */
	p_sb = kmalloc(sizeof(*p_sb), GFP_KERNEL);
-	if (!p_sb) {
-		DP_NOTICE(cdev, "Failed to allocate `struct qed_sb_attn_info'\n");
+	if (!p_sb)
		return -ENOMEM;
-	}

	/* SB ring  */
	p_virt = dma_alloc_coherent(&cdev->pdev->dev,
@@ -2381,7 +2379,6 @@ static int qed_int_sb_attn_alloc(struct
				    &p_phys, GFP_KERNEL);

	if (!p_virt) {
-		DP_NOTICE(cdev, "Failed to allocate status block (attentions)\n");
		kfree(p_sb);
		return -ENOMEM;
	}
@@ -2667,17 +2664,14 @@ static int qed_int_sp_sb_alloc(struct qe

	/* SB struct */
	p_sb = kmalloc(sizeof(*p_sb), GFP_KERNEL);
-	if (!p_sb) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `struct qed_sb_info'\n");
+	if (!p_sb)
		return -ENOMEM;
-	}

	/* SB ring  */
	p_virt = dma_alloc_coherent(&p_hwfn->cdev->pdev->dev,
				    SB_ALIGNED_SIZE(p_hwfn),
				    &p_phys, GFP_KERNEL);
	if (!p_virt) {
-		DP_NOTICE(p_hwfn, "Failed to allocate status block\n");
		kfree(p_sb);
		return -ENOMEM;
	}
@@ -2959,7 +2953,6 @@ int qed_int_igu_read_cam(struct qed_hwfn
	u16 prev_sb_id = 0xFF;

	p_hwfn->hw_info.p_igu_info = kzalloc(sizeof(*p_igu_info), GFP_KERNEL);
-
	if (!p_hwfn->hw_info.p_igu_info)
		return -ENOMEM;

@@ -3121,18 +3114,14 @@ int qed_int_alloc(struct qed_hwfn *p_hwf
	int rc = 0;

	rc = qed_int_sp_dpc_alloc(p_hwfn);
-	if (rc) {
-		DP_ERR(p_hwfn->cdev, "Failed to allocate sp dpc mem\n");
+	if (rc)
		return rc;
-	}
+
	rc = qed_int_sp_sb_alloc(p_hwfn, p_ptt);
-	if (rc) {
-		DP_ERR(p_hwfn->cdev, "Failed to allocate sp sb mem\n");
+	if (rc)
		return rc;
-	}
+
	rc = qed_int_sb_attn_alloc(p_hwfn, p_ptt);
-	if (rc)
-		DP_ERR(p_hwfn->cdev, "Failed to allocate sb attn mem\n");

	return rc;
 }
Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_main.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_main.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_main.c
@@ -841,10 +841,8 @@ static int qed_slowpath_start(struct qed
	if (IS_PF(cdev)) {
		/* Allocate stream for unzipping */
		rc = qed_alloc_stream_mem(cdev);
-		if (rc) {
-			DP_NOTICE(cdev, "Failed to allocate stream memory\n");
+		if (rc)
			goto err2;
-		}

		/* First Dword used to diffrentiate between various sources */
		data = cdev->firmware->data + sizeof(u32);
Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_mcp.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_mcp.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_mcp.c
@@ -171,7 +171,6 @@ int qed_mcp_cmd_init(struct qed_hwfn *p_
	return 0;

 err:
-	DP_NOTICE(p_hwfn, "Failed to allocate mcp memory\n");
	qed_mcp_free(p_hwfn);
	return -ENOMEM;
 }
Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_spq.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_spq.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_spq.c
@@ -323,10 +323,8 @@ struct qed_eq *qed_eq_alloc(struct qed_h

	/* Allocate EQ struct */
	p_eq = kzalloc(sizeof(*p_eq), GFP_KERNEL);
-	if (!p_eq) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `struct qed_eq'\n");
+	if (!p_eq)
		return NULL;
-	}

	/* Allocate and initialize EQ chain*/
	if (qed_chain_alloc(p_hwfn->cdev,
@@ -335,10 +333,8 @@ struct qed_eq *qed_eq_alloc(struct qed_h
			    QED_CHAIN_CNT_TYPE_U16,
			    num_elem,
			    sizeof(union event_ring_element),
-			    &p_eq->chain)) {
-		DP_NOTICE(p_hwfn, "Failed to allocate eq chain\n");
+			    &p_eq->chain))
		goto eq_allocate_fail;
-	}

	/* register EQ completion on the SP SB */
	qed_int_register_cb(p_hwfn, qed_eq_completion,
@@ -451,10 +447,8 @@ int qed_spq_alloc(struct qed_hwfn *p_hwf

	/* SPQ struct */
	p_spq = kzalloc(sizeof(struct qed_spq), GFP_KERNEL);
-	if (!p_spq) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `struct qed_spq'\n");
+	if (!p_spq)
		return -ENOMEM;
-	}

	/* SPQ ring  */
	if (qed_chain_alloc(p_hwfn->cdev,
@@ -463,18 +457,14 @@ int qed_spq_alloc(struct qed_hwfn *p_hwf
			    QED_CHAIN_CNT_TYPE_U16,
			    0,   /* N/A when the mode is SINGLE */
			    sizeof(struct slow_path_element),
-			    &p_spq->chain)) {
-		DP_NOTICE(p_hwfn, "Failed to allocate spq chain\n");
+			    &p_spq->chain))
		goto spq_allocate_fail;
-	}

	/* allocate and fill the SPQ elements (incl. ramrod data list) */
	capacity = qed_chain_get_capacity(&p_spq->chain);
	p_virt = dma_alloc_coherent(&p_hwfn->cdev->pdev->dev,
-				    capacity *
-				    sizeof(struct qed_spq_entry),
+				    capacity * sizeof(struct qed_spq_entry),
				    &p_phys, GFP_KERNEL);
-
	if (!p_virt)
		goto spq_allocate_fail;

@@ -863,10 +853,8 @@ struct qed_consq *qed_consq_alloc(struct

	/* Allocate ConsQ struct */
	p_consq = kzalloc(sizeof(*p_consq), GFP_KERNEL);
-	if (!p_consq) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `struct qed_consq'\n");
+	if (!p_consq)
		return NULL;
-	}

	/* Allocate and initialize EQ chain*/
	if (qed_chain_alloc(p_hwfn->cdev,
@@ -874,10 +862,8 @@ struct qed_consq *qed_consq_alloc(struct
			    QED_CHAIN_MODE_PBL,
			    QED_CHAIN_CNT_TYPE_U16,
			    QED_CHAIN_PAGE_SIZE / 0x80,
-			    0x80, &p_consq->chain)) {
-		DP_NOTICE(p_hwfn, "Failed to allocate consq chain");
+			    0x80, &p_consq->chain))
		goto consq_allocate_fail;
-	}

	return p_consq;

Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_sriov.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_sriov.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_sriov.c
@@ -455,10 +455,8 @@ int qed_iov_alloc(struct qed_hwfn *p_hwf
	}

	p_sriov = kzalloc(sizeof(*p_sriov), GFP_KERNEL);
-	if (!p_sriov) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `struct qed_sriov'\n");
+	if (!p_sriov)
		return -ENOMEM;
-	}

	p_hwfn->pf_iov_info = p_sriov;

@@ -507,10 +505,9 @@ int qed_iov_hw_info(struct qed_hwfn *p_h

	/* Allocate a new struct for IOV information */
	cdev->p_iov_info = kzalloc(sizeof(*cdev->p_iov_info), GFP_KERNEL);
-	if (!cdev->p_iov_info) {
-		DP_NOTICE(p_hwfn, "Can't support IOV due to lack of memory\n");
+	if (!cdev->p_iov_info)
		return -ENOMEM;
-	}
+
	cdev->p_iov_info->pos = pos;

	rc = qed_iov_pci_cfg_info(cdev);
Index: compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_vf.c
===================================================================
--- compat-rdma-work.orig/drivers/net/ethernet/qlogic/qed/qed_vf.c
+++ compat-rdma-work/drivers/net/ethernet/qlogic/qed/qed_vf.c
@@ -331,31 +331,23 @@ int qed_vf_hw_prepare(struct qed_hwfn *p

	/* Allocate vf sriov info */
	p_iov = kzalloc(sizeof(*p_iov), GFP_KERNEL);
-	if (!p_iov) {
-		DP_NOTICE(p_hwfn, "Failed to allocate `struct qed_sriov'\n");
+	if (!p_iov)
		return -ENOMEM;
-	}

	/* Allocate vf2pf msg */
	p_iov->vf2pf_request = dma_alloc_coherent(&p_hwfn->cdev->pdev->dev,
						  sizeof(union vfpf_tlvs),
						  &p_iov->vf2pf_request_phys,
						  GFP_KERNEL);
-	if (!p_iov->vf2pf_request) {
-		DP_NOTICE(p_hwfn,
-			  "Failed to allocate `vf2pf_request' DMA memory\n");
+	if (!p_iov->vf2pf_request)
		goto free_p_iov;
-	}

	p_iov->pf2vf_reply = dma_alloc_coherent(&p_hwfn->cdev->pdev->dev,
						sizeof(union pfvf_tlvs),
						&p_iov->pf2vf_reply_phys,
						GFP_KERNEL);
-	if (!p_iov->pf2vf_reply) {
-		DP_NOTICE(p_hwfn,
-			  "Failed to allocate `pf2vf_reply' DMA memory\n");
+	if (!p_iov->pf2vf_reply)
		goto free_vf2pf_request;
-	}

	DP_VERBOSE(p_hwfn,
		   QED_MSG_IOV,
